### PR TITLE
Backend: dummy data generation

### PR DESCRIPTION
**Pull request by kobim**
A rake task to generate dummy data.
Takes 2 parameters: number of restaurants and number of reviews per each restaurant (defaults to 10 and 5).

### CODE REVIEW

File: server/lib/tasks/generate_data.rake [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    restaurant = Restaurant.new(
+      name: Faker::Company.name,
+      cuisine: Restaurant::CUISINES.sample,
+      rating: rand(0.0..5.0).round(1),
+      tenbis: [true, false].sample,
+      address: Faker::Address.full_address,
+      delivery_time: rand(30..120)
+    )
+    number_of_reviews.times do
+      restaurant.reviews << Review.new(
```
Comment (itaizabWW) [inline server/lib/tasks/generate_data.rake]: maybe you can just put the restaurant id on the review

File: server/lib/tasks/generate_data.rake [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  Rails.logger = Logger.new(STDOUT, level: :info)
+
+  Rails.logger.info "Generating #{number_of_restaurants} fake restaurants and #{number_of_reviews} reviews"
+
+  number_of_restaurants.times do
+    restaurant = Restaurant.new(
+      name: Faker::Company.name,
+      cuisine: Restaurant::CUISINES.sample,
+      rating: rand(0.0..5.0).round(1),
```
Comment (itaizabWW) [inline server/lib/tasks/generate_data.rake]: minor -you can also use `rand(5)` instead

File: server/lib/tasks/generate_data.rake [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      tenbis: [true, false].sample,
+      address: Faker::Address.full_address,
+      delivery_time: rand(30..120)
+    )
+    number_of_reviews.times do
+      restaurant.reviews << Review.new(
+        name: Faker::Name.name,
+        rating: rand(1.0..5.0).round(1),
```
Comment (itaizabWW) [inline server/lib/tasks/generate_data.rake]: same as above :)

File: server/lib/tasks/generate_data.rake [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        comment: Faker::ChuckNorris.fact
+      )
+    end
+    restaurant.save!
+  end
+
+  Rails.logger.info 'Done!'
```
Comment (itaizabWW) [inline server/lib/tasks/generate_data.rake]: maybe we can add more info for the logging - something like:
```
Rails.logger.info("Successfully generated #{number_of_restaurants} fake restaurants and #{number_of_reviews} reviews") 
```

File: server/lib/tasks/generate_data.rake [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      delivery_time: rand(30..120)
+    )
+    number_of_reviews.times do
+      restaurant.reviews << Review.new(
+        name: Faker::Name.name,
+        rating: rand(1.0..5.0).round(1),
+        comment: Faker::ChuckNorris.fact
+      )
+    end
+    restaurant.save!
```
Comment (itaizabWW) [inline server/lib/tasks/generate_data.rake]: minor- maybe you want to catch errors and log an error with an informative message like:
```
"Failed to generate fake restaurants..."
```

File: server/lib/tasks/generate_data.rake (hunk 2 above)
Comment (kobim) [reply server/lib/tasks/generate_data.rake]: I need a float, between 0 and 5 (inclusive). Hence I used the range syntax and not `rand(5.0)`

File: server/lib/tasks/generate_data.rake [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      end
+    rescue StandardError => e
+      Rails.logger.error "Exception: #{e.message}, rolled back."
+    else
+      Rails.logger.info "Successfully generated #{number_of_restaurants} restaurants " \
+                        "with total of #{number_of_reviews} reviews!"
+    end
+  end
+end
+
```
Comment (NoyaArie) [inline server/lib/tasks/generate_data.rake]: please add `private`

**Pull request merged**